src/bin/taker.rs

```diff
@@ -1,3 +1,183 @@
+use std::path::PathBuf;
+
+use bitcoind::bitcoincore_rpc::{json::ListUnspentResultEntry, Auth};
+use clap::Parser;
+use coinswap::{
+    taker::{SwapParams, Taker, TakerBehavior},
+    utill::{
+        parse_proxy_auth, read_bitcoin_network_string, read_connection_network_string, setup_logger,
+    },
+    wallet::RPCConfig,
+};
+
+#[derive(Parser, Debug)]
+#[clap(version = option_env ! ("CARGO_PKG_VERSION").unwrap_or("unknown"),
+author = option_env ! ("CARGO_PKG_AUTHORS").unwrap_or(""))]
+struct Cli {
+    #[clap(long, default_value = "clearnet",possible_values = &["tor","clearnet"])]
+    network: String,
+    #[clap(long, short = 'd')]
+    data_directory: Option<PathBuf>,
```

Comment:
Adding code comments on the elements is an easy way to produce cli help doc. A bit of help doc is needed before publishing this. Doesn't have to be very neat.